### PR TITLE
Sync operator apis

### PR DESCRIPTION
**Pull request by avivgold098**
Hi,

In order to support the version of prometheus-operator v0.40.0, I updated the relevant types.

### CODE REVIEW

**Comment by avivgold098** [comment]
The prometheus-operator api https://github.com/coreos/prometheus-operator/blob/v0.40.0/Documentation/api.md

File: v1/EmbeddedObjectMetadata.dhall [hunk 1]
```diff
@@ -0,0 +1,15 @@
+let imports = ../imports.dhall
+
+let Map = imports.Prelude.Map.Type
+
+in  { Type =
+        { name : Optional Text
+        , lables : Optional (Map Text Text)
```
Comment (ari-becker) [inline v1/EmbeddedObjectMetadata.dhall:7]: `labels`

File: v1/StorageSpec.dhall [hunk 2]
```diff
@@ -2,11 +2,16 @@ let imports = ../imports.dhall
...
 
-let VolumeClaimTemplate = ./VolumeClaimTemplate.dhall
+let EmbeddedPersistentVolumeClaim = ./EmbeddedPersistentVolumeClaim.dhall
 
-let StorageSpec =
-      < EmptyDir : { emptyDir : Kubernetes.EmptyDirVolumeSource.Type }
-      | VolumeClaimTemplate : { volumeClaimTemplate : VolumeClaimTemplate.Type }
-      >
-
-in  StorageSpec
```
Comment (ari-becker) [inline v1/StorageSpec.dhall:12]: Why did you remove this? A `StorageSpec` need to be either an `emptyDir` or a `volumeClaimTemplate`. Please return it

File: v1/EmbeddedObjectMetadata.dhall (hunk 1 above)
Comment (avivgold098) [reply v1/EmbeddedObjectMetadata.dhall:7]: HI it will  be in a separate PR

Comment (avivgold098) [reply v1/EmbeddedObjectMetadata.dhall:7]: it will be in a separate pr

File: v1/StorageSpec.dhall (hunk 2 above)
Comment (avivgold098) [reply v1/StorageSpec.dhall]: fixed
